language/tools/mirai-dataflow-analysis/src/ddlog_output.rs

```diff
@@ -0,0 +1,100 @@
+// Copyright (c) The Diem Core Contributors
+// SPDX-License-Identifier: Apache-2.0
+
+use regex::Regex;
+use std::{
+    fs,
+    io::Write,
+    path::Path,
+    process::{Command, Stdio},
+};
+
+use crate::{
+    datalog::{determine_op_type, DatalogRelation, DatalogRelationOperand, DatalogRelations},
+    util::get_child_output,
+};
+
+// Run the Differential Datalog analysis on the test file and
+// capture output
+pub fn run_ddlog_analysis(
+    ddlog_output_path: &Path,
+    analysis_output_path: &Path,
+) -> Result<(), String> {
+    let ddlog_cli_path = "analyses/ddlog_ddlog/target/release/ddlog_cli";
```

Comment:
Consider making `"analyses/ddlog_ddlog/target/release/ddlog_cli"` a global constant

Reply:
Also, what is the assumption about running this path? Must it be under the output dir?

Reply:
This path will actually be automatically created as part of the Differential Datalog setup; so it should exist exactly as above unless the analysis is moved from the `analyses` folder.

Reply:
Ok, in this case, it is a judgement call to make it a constant, but will be good to at least put a comment there saying that this path is auto-generated by `Differential Datalog`.

Reply:
I'll make this a constant and add a comment, same thing on the `souffle_output` side.